### DISCUSSION TITLE
Rest api

### DISCUSSION THREAD

**Discussion by linusjf** [marcomontalbano/video-to-markdown#64] [General]
Do you have a rest api for this?

**Comment by linusjf** [comment]
Why can't I just use the YouTube thumbnail images and generate the markdown from that?

**Reply by marcomontalbano** [reply]
Yes, you can do it by yourself or you can deselect the "Show play icon". This way, I'll directly generate the markdown using the youtube thumbnail without attaching the play icon.

This is an example of the markdown you'll get:
```md
[![[Big Buck Bunny](https://www.youtube.com/watch?v=aqz-KE-bpKQ)](https://i.ytimg.com/vi/aqz-KE-bpKQ/maxresdefault.jpg)](https://www.youtube.com/watch?v=aqz-KE-bpKQ "")
```

**Reply by linusjf** [reply]
I'm not familiar with Cloudinaire. But from the little I read, it appears as though the entire video has to be uploaded to their site to generate thumbnails via jpg urls. Is that correct and that's how you do it as well? Also, is it possible to simply access the Cloudinaire api to generate thumbnails instead of via your site?

**Reply by linusjf** [reply]
Additionally, not all videos on YouTube seem to have thumbnails.

**Reply by linusjf** [reply]
I don't think you actually upload the video to Cloudinaire. It would take way too long.

**Reply by marcomontalbano** [reply]
No, I'm not uploading videos to Cloudinary at all! I'm using Cloudinary APIs to attach the play icon to the video thumbnail.
Taking YouTube as an example, from the following thumbnail:
```md
https://i.ytimg.com/vi/aqz-KE-bpKQ/maxresdefault.jpg
```
![Big Buck Bunny](https://i.ytimg.com/vi/aqz-KE-bpKQ/maxresdefault.jpg)

You'll get this:
```md
https://res.cloudinary.com/marcomontalbano/image/upload/v1626375833/video_to_markdown/images/youtube--aqz-KE-bpKQ-c05b58ac6eb4c4700831b2b3070cd403.jpg
```
![Big Buck Bunny](https://res.cloudinary.com/marcomontalbano/image/upload/v1626375833/video_to_markdown/images/youtube--aqz-KE-bpKQ-c05b58ac6eb4c4700831b2b3070cd403.jpg)

You can absolutely do that same using Cloudinary API (also without Cloudinary), this tool aims to provide a faster way to achieve the result just by passing a video URL.

I'll add an "How it work!" section to betterer clarify all these steps 😄

**Reply by marcomontalbano** [reply]
> Additionally, not all videos on YouTube seem to have thumbnails.

If you find one, let me know 👍

**Comment by marcomontalbano** [comment]
Hi @Fernal73, I'm not planning to add a dedicated API endpoint at the moment, but I don't exclude the possibility in the future.

**Comment by linusjf** [comment]
https://github.com/Fernal73/CIAY/blob/main/January/Day005.md

This one, unless I missed something.

**Reply by marcomontalbano** [reply]
Here it is 😄  https://i.ytimg.com/vi/XscjbW08sHk/maxresdefault.jpg

Using [video-to-markdown](https://video-to-markdown.marcomontalbano.com/) you just need to paste the YouTube video link and press a button.
You'll get the markdown in a second.


https://user-images.githubusercontent.com/1681269/217771176-2381164c-6163-4d38-bec2-b8f88145af21.mov

**Reply by linusjf** [reply]
I want to be able to generate those programmatically.

**Reply by linusjf** [reply]
If I understand your process, you've uploaded the thumbprint to Cloudinaire, use the transformation api to overlay this image with the play icon and provide the new transformed image as the img link in the markdown. The other way of doing this is probably via CSS which doesn't work well with markdown and is ugly as hell.

**Reply by marcomontalbano** [reply]
Yes, the process you described is correct.

Regarding the thumbnail, given a video like this one https://www.youtube.com/watch?v=aqz-KE-bpKQ, you just need to extract the video id from the URL that in the example is `aqz-KE-bpKQ` and then you can use this id to compose the thumbnail URL: `https://i.ytimg.com/vi/<VIDEO_ID>/maxresdefault.jpg`.
In the end, you'll get the following: https://i.ytimg.com/vi/aqz-KE-bpKQ/maxresdefault.jpg

**Reply by linusjf** [reply]
I think ImageMagick works for me. GIMP might do it as well.
